src/utilities/extendSchema.js

```diff
@@ -72,10 +73,7 @@ export function extendSchema(
   documentAST: DocumentNode,
   options?: Options,
 ): GraphQLSchema {
-  invariant(
-    schema instanceof GraphQLSchema,
-    'Must provide valid GraphQLSchema',
-  );
+  assertValidSchema(schema);
```

Comment:
Oh good call to check this here - but I'm not sure we want to assert the whole schema is valid before extending it, we want to preserve the use case where you can extend an incomplete schema to get to a complete one.

I have an idea to extract just this check out, one sec while I test it out